hypervisor/include/arch/x86/assign.h

```diff
@@ -9,18 +9,149 @@
 
 #include <ptdev.h>
 
-void ptdev_intx_ack(struct acrn_vm *vm, uint8_t virt_pin,
-		enum ptdev_vpin_source vpin_src);
-int ptdev_msix_remap(struct acrn_vm *vm, uint16_t virt_bdf,
-		uint16_t entry_nr, struct ptdev_msi_info *info);
-int ptdev_intx_pin_remap(struct acrn_vm *vm, uint8_t virt_pin,
-		enum ptdev_vpin_source vpin_src);
-int ptdev_add_intx_remapping(struct acrn_vm *vm, uint8_t virt_pin, uint8_t phys_pin,
-		bool pic_pin);
+/**
+ * @file assign.h
+ *
+ * @brief public APIs for Passthrough Interrupt Remapping
+ */
+
+/**
+ * @brief VT-d
+ *
+ * @defgroup acrn_passthrough ACRN Passthrough
+ * @{
+ */
+
+/**
+ * @brief Acknowledge a virtual interrupt for passthrough device.
+ *
+ * Acknowledge a virtual legacy interrupt for a passthrough device.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_pin virtual pin number associated with the passthrough device
+ * @param[in] vpin_src ioapic or pic
+ *
+ * @return None
+ *
+ * @pre vm != NULL
+ *
+ */
+void ptdev_intx_ack(struct acrn_vm *vm, uint8_t virt_pin, enum ptdev_vpin_source vpin_src);
+
+/**
+ * @brief MSI/MSI-x remapping for passthrough device.
+ *
+ * Main entry for PCI device assignment with MSI and MSI-X.
+ * MSI can up to 8 vectors and MSI-X can up to 1024 Vectors.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_bdf virtual bdf associated with the passthrough device
+ * @param[in] entry_nr indicate coming vectors, entry_nr = 0 means first vector
+ * @param[in] info structure used for MSI/MSI-x remapping
+ *
+ * @return 0 - on success
+ * @return -ENODEV - for SOS, the entry already be held by others; for UOS, no pre-hold mapping found.
+ *
+ * @pre vm != NULL
+ * @pre info != NULL
+ *
+ */
+int ptdev_msix_remap(struct acrn_vm *vm, uint16_t virt_bdf, uint16_t entry_nr, struct ptdev_msi_info *info);
+
+
+/**
+ * @brief INTx remapping for passthrough device.
+ *
+ * Set up the remapping of the given virtual pin for the given vm.
+ * Thsi is the main entry for PCI/Legacy device assignment with INTx, calling from vIOAPIC or vPIC.
+ *
+ * @param[in] vm pointer to acrn_vm
+ * @param[in] virt_pin virtual pin number associated with the passthrough device
+ * @param[in] vpin_src ioapic or pic
+ *
+ * @return 0 - on success
+ * @return -ENODEV - for SOS, the entry already be held by others; for UOS, no pre-hold mapping found.
```

Comment:
Same comment as above, can you use bullet lists here instead of multiple `@return` lines?